lib/src/err/mod.rs

```diff
@@ -107,6 +107,13 @@ pub enum Error {
 		message: String,
 	},
 
+	/// Given test operation failed for JSON Patch
+	#[error("Given test operation failed for JSON Patch. Expected `{expected}`, but got `{got}` instead.")]
+	PatchTestFailure {
+		expected: Value,
+		got: Value,
+	},
```

Comment:
Could we rename this to `PatchTest`? That will make it `Error::PatchTest` which removes the redundant 'Failure'!

Reply:
Indeed, I've been anticipating a review of this, as naming things isn't my strong suit 😅